File: 70-climbing-stairs/climbing-stairs.java
class Solution {
    public int solve(int n , int stairs, int [] dp){
        if(stairs == n){
            return 1;
        }
        if(stairs > n ){
            return 0;
        }
        if(dp[stairs]!=-1){
            return dp[stairs];
        }

        return dp[stairs] = solve(n, stairs+1, dp ) + solve(n, stairs+2 , dp);

    }

    public int climbStairs(int n) {
        // simple recursive method - > TLE 
        int stairs = 0 ;
        int dp []  = new int [n+1];
        Arrays.fill(dp , -1);
        return solve(n , stairs , dp);

    }
}

File: 70-climbing-stairs/climbing-stairs.py
class Solution:

    def solve(self, n:int , dp:List[int]) ->int:
        if(n ==0 or  n== 1):
            return 1
        elif(n==2):
            return 2
        elif(dp[n]!=-1):
            return dp[n]
        else:
            dp[n] = (self.solve(n-1, dp) + self.solve(n-2, dp))
        return dp[n]

    def climbStairs(self, n: int) -> int:
        dp = [-1]*(n+1)

        if(dp[n] != -1):
            return dp[n]
        else:
            return self.solve(n ,dp)